ksqldb-engine/src/main/java/io/confluent/ksql/query/TransactionAuthorizationClassifier.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query;
+
+import io.confluent.ksql.query.QueryError.Type;
+import java.util.Objects;
+import org.apache.kafka.common.errors.ClusterAuthorizationException;
+import org.apache.kafka.common.errors.TransactionalIdAuthorizationException;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@code TransactionAuthorizationClassifier} classifies missing cluster ACLs as user error
+ */
+public class TransactionAuthorizationClassifier implements QueryErrorClassifier {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TransactionAuthorizationClassifier.class);
+
+  private final String queryId;
+
+  public TransactionAuthorizationClassifier(final String queryId) {
+    this.queryId = Objects.requireNonNull(queryId, "queryId");
+  }
+
+  @Override
+  public Type classify(final Throwable e) {
+    final Type type =
+        e instanceof StreamsException
```

Comment:
@guozhangwang It's unclear to me, why this exception is wrapped with `StreamsException` while `TopicAuthorizationException` and `GroupAuthorizationException` are not?

For `MissingSourceTopicException` it's clear because it extends `StreamsException` and is thrown by KS directly.

Reply:
I think it should not, and it should be fixed via https://cwiki.apache.org/confluence/display/KAFKA/KIP-691%3A+Enhance+Transactional+Producer+Exception+Handling

cc @abbccdda

Reply:
But the producer should not wrap anything with `StreamsException` -- so why does KS wrap the one but not the other producer exception? And how would the KIP change it (it seem the KIP is for the producer but not for KS)?

Reply:
`TopicAuthorizationException`: this can be thrown from both consumer and producer, and for consumer it could throw from `partitionsFor` which is used in KS via global store, in that place we just do not wrap it and let it throw all the way up. For producer it may be thrown from the callback, in which we do wrap --- this is an inconsistency.

`GroupAuthorizationException`: only thrown from producer in latest version with KIP-447, for which we should wrap in KS; in older version it could be thrown from consumer and again we do not wrap in KS.

`TransactionalIdAuthorizationException`: thrown from producer, in the callback where we always wrap.

Reply:
In our other classifiers (e.g. the regex classifier we walk the cause stack and try and find any exception that matches). That would be an approach to make this check more robust.